setup

```diff
@@ -26,17 +27,34 @@ HELP_MESSAGE = """\
     For usage info run: python3 setup -h
     """
 
+class BuildType(Enum):
+    DEBUG = "Debug"
+    RELEASE = "Release"
+
+    def __str__(self):
+        return self.value
+
 
 def get_arguments():
     parser = argparse.ArgumentParser(
         description="MAGE setup script",
     )
     subparsers = parser.add_subparsers(help="sub-command help", dest="action")
 
-    build_parser = subparsers.add_parser("build", help="Build memgraph-mage")
-    build_parser.add_argument(
+    build_args_parser = argparse.ArgumentParser(description="Build arguments", add_help=False)
+    build_args_parser.add_argument(
         "-p", "--path", help="Path to query modules directory", required=False
     )
+    build_args_parser.add_argument(
+        "--type", help="Build type", type=BuildType, choices=list(BuildType), required=False
+    )
+    build_args_parser.add_argument(
+        "--gpu", help="GPU Algorithms", action='store_true'
+    )
+    build_args_parser.add_argument(
+        "--cpp-build-flags", nargs="+", help="CMake flags for the cpp part (without -D)", required=False
+    )
```

Comment:
Maybe instead of `--gpu` flag we can use this `cpp` flags to build cuGraph, so the user will call the script with:
`python3 --cpp-build-flags MAGE_CUGRAPH_ENABLE=ON ...`

Reply:
This `MAGE_CUGRAPH_ENABLE` is nonetheless `cpp` flag, so it will be easier for the user maybe?

Reply:
Okay you are right haha